api/handlers_mobile.go

```diff
@@ -784,3 +793,91 @@ func MobileSearchHandler(c *gin.Context) {
 		NewInternalError(err).Abort(c)
 	}
 }
+
+// call feed api to get results like search - https://kabbalahmedia.info/feed_api/feed
+// write sql to get
+func MobileFeed(c *gin.Context){
+	// get input json
+	var r MobileFeedRequest
+	if c.Bind(&r) != nil {
+		return
+	}
+
+	feedInputJson, err := json.Marshal(r)
+	if err != nil {
+		NewInternalError(err).Abort(c)
+	}
+
+	feedApi, err := getFeedApi("feed")
+	if err != nil {
+		NewInternalError(err).Abort(c)
+	}
+
+	feedResponseObj, err := http.Post(feedApi, "application/json", strings.NewReader(string(feedInputJson)))
+	if err != nil {
+		NewInternalError(err).Abort(c)
+	}
+
+	// convert response body to byte arr and then get the feed array
+	feedRespBytes, err := io.ReadAll(feedResponseObj.Body)
+	if err != nil {
+		log.Error(err.Error())
+	}
+
+	feedBody := new(feedResponseType)
+	if err = json.Unmarshal(feedRespBytes, feedBody); err != nil || feedBody == nil {
+		log.Error(err.Error())
+	}
+
+	imagesUrlTemplate := viper.GetString("content_unit_images.url_template")
+	var mobilefeedResponse []*MobileFeedResponseItem
+
+	db := c.MustGet("MDB_DB").(*sql.DB)
+	language := c.Query("language");
+
+	var cuIds []string
+	itemsMap := make(map[string]*MobileFeedResponseItem)
+
+	for _, item := range feedBody.Feed {
+		imageStr := fmt.Sprintf(imagesUrlTemplate, item.ContentUnitUid)
+
+		feedResp := &MobileFeedResponseItem {
+			ContentUnitUid: item.ContentUnitUid,
+			Type: item.ContentType,
+			Image: &imageStr,
+			Date: item.Date,
+			// Views: ,
+			Title: getContentUnitName(item.ContentUnitUid, language, db),
```

Comment:
you call DB from loop its not good IMHO. put all CU uids to slice and take it with one DB call

Reply:
done